llms/mlx_lm/server.py

```diff
@@ -602,7 +665,7 @@ def handle_chat_completions(self) -> mx.array:
             prompt = convert_chat(body["messages"], body.get("role_mapping"))
             prompt = self.tokenizer.encode(prompt)
 
-        return mx.array(prompt)
+        return prompt
```

Comment:
So this returns a list now? I am not sure why although there is no need to cast to mlx too early I guess. At the very least we probably need to change the return type annotation and docstring.